### PR TITLE
Relationship type now using the new api structure

### PR DESCRIPTION
**Pull request by dave-cheong**

### CODE REVIEW

File: backend/typescript/controllers/helpers.ts [hunk 1]
```diff
@@ -7,34 +7,38 @@ export function sendResource<T>(res: Response) {
     return (doc: T): T => {
         if (doc) {
             res.json(doc);
+            res.status(200);
```
Comment (sundriver) [inline backend/typescript/controllers/helpers.ts]: does this actually work, I feel json sends 200 as part of the request, 
like this : res.status(500).json({ error: 'message' });

File: backend/typescript/controllers/helpers.ts [hunk 2]
```diff
@@ -115,9 +119,9 @@ export function sendError<T>(res: Response) {
 export function sendNotFoundError<T>(res: Response) {
     'use strict';
     return (doc: T): T => {
-        res.status(404);
         if (!doc) {
             res.json(new ErrorResponse(404, 'Can\'t find the requested resource.'));
+            res.status(404);
```
Comment (sundriver) [inline backend/typescript/controllers/helpers.ts]: same as above
also removing error code from error response

File: backend/typescript/seeding/seed.ts [hunk 3]
```diff
@@ -48,7 +48,9 @@ class Seeder {
     }
 
     /* tslint:disable:max-func-body-length */
-    public static async createRelationshipTypeModel(values:IRelationshipType,attributeNames:IRelationshipAttributeName[]) {
+    public static async createRelationshipTypeModel
```
Comment (sundriver) [inline backend/typescript/seeding/seed.ts]: I love to see this method split into two method.
It just does too much for a single method

File: backend/typescript/controllers/helpers.ts (hunk 1 above)
Comment (dave-cheong) [inline backend/typescript/controllers/helpers.ts]: Yeah it works in that order too. Without setting it, it was always setting the status to 404 by the catch all.

File: backend/typescript/seeding/seed.ts (hunk 3 above)
Comment (dave-cheong) [inline backend/typescript/seeding/seed.ts]: I split it out to two functions.

File: backend/typescript/controllers/helpers.ts (hunk 2 above)
Comment (dave-cheong) [inline backend/typescript/controllers/helpers.ts]: Ok removed the status code from the ErrorResponse and the client should check the status from the HTTP response.

**Pull request merged**